src/erlcloud_sns.erl

```diff
@@ -352,8 +354,8 @@ list_subscriptions_by_topic(TopicArn, NextToken, Config) when is_record(Config,
             Doc),
     Decoded.
 
--spec list_subscriptions_by_topic_all/1 :: (string()) -> proplist().
--spec list_subscriptions_by_topic_all/2 :: (string(), aws_config()) -> proplist().
+-spec list_subscriptions_by_topic_all/1 :: (parameter()) -> proplist().
+-spec list_subscriptions_by_topic_all/2 :: (parameter(), aws_config()) -> proplist().
```

Comment:
Line 358 is too long: -spec list_subscriptions_by_topic_all/2 :: (parameter(), aws_config()) -> proplist()..